src/gc/handletablecore.cpp

```diff
@@ -516,14 +516,14 @@ BOOL SegmentInitialize(TableSegment *pSegment, HandleTable *pTable)
 
 #ifndef FEATURE_REDHAWK // todo: implement SafeInt
     // Prefast overflow sanity check the addition
-    if (!ClrSafeInt<uint32_t>::addition(dwCommit, g_SystemInfo.dwPageSize, dwCommit))
+    if (!ClrSafeInt<uint32_t>::addition(dwCommit, OS_PAGE_SIZE, dwCommit))
```

Comment:
This needs to be the real OS size determined at runtime since that's the granularity with which the memory can be committed. For example, the ARM64 Linux distros have varying page size.
The same is true for all other places below where you have replaced the g_SystemInfo.dwPageSize with OS_PAGE_SIZE.